src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -2388,28 +2426,59 @@ private StringBuilder GetTextWithoutPrompts(StringBuilder builder, SnapshotSpan
                     {
                         continue;
                     }
-                    if (!IsPrompt(sourceSpan))
+                    var sourceSnapshot = sourceSpan.Snapshot;
+                    var mappedSpans = TextView.BufferGraph.MapDownToBuffer(span, SpanTrackingMode.EdgeExclusive, sourceSnapshot.TextBuffer);
+                    bool added = false;
+                    foreach (var mappedSpan in mappedSpans)
                     {
-                        var sourceSnapshot = sourceSpan.Snapshot;
-                        var mappedSpans = TextView.BufferGraph.MapDownToBuffer(span, SpanTrackingMode.EdgeExclusive, sourceSnapshot.TextBuffer);
-                        bool added = false;
-                        foreach (var mappedSpan in mappedSpans)
+                        var intersection = sourceSpan.Span.Intersection(mappedSpan);
+                        if (intersection.HasValue && !intersection.Value.IsEmpty)
```

Comment:
Empty intersections should only occur at the end of the session (input prompt but no input), and the resulting block will be ignored by paste.